react/components/ProductImage.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { pathOr, isEmpty } from 'ramda'
+import PropTypes from 'prop-types'
+import {
+  CollectionBadges,
+  DiscountBadge,
+} from 'vtex.store-components'
+
+import Image from './Image'
+import { productShape } from '../utils/propTypes'
+
+import productSummary from '../productSummary.css'
+
+const ProductImage = ({ product, showBadge, badgeText, showCollections }) => {
+  const {
+    productClusters,
+    productName: name,
+    sku: {
+      image: { imageUrl },
+    },
+  } = product
+
+  const commertialOffer = pathOr({}, ['sku', 'seller', 'commertialOffer'], product)
+
+  const withBagde = img => (
```

Comment:
Wouldn't be better that these internal functions be defined outside the `ProductImage` component? You would have to make some changes because each function would have to receive their conditionals, but would make this component much cleaner.

Reply:
It would make the functions outside `ProductImage` much more complex. They would need not only receive their conditionals but some other props used to create the used components, such as `productClusters` in `withCollection`

Reply:
Yean, but they are already some kinda of components, so it would make sense that they get a bit complex.